src/SAML2/XML/saml/AttributeValue.php

```diff
@@ -106,13 +100,27 @@ public static function fromXML(DOMElement $xml): static
         Assert::same($xml->localName, 'AttributeValue', InvalidDOMElementException::class);
         Assert::same($xml->namespaceURI, AttributeValue::NS, InvalidDOMElementException::class);
 
-        $value = $xml->textContent;
-        if (
+        if ($xml->childElementCount > 0) {
+            $node = $xml->firstElementChild;
+
+            if (str_contains($node->tagName, ':')) {
+                list($prefix, $eltName) = explode(':', $node->tagName);
+                $className = sprintf('\SimpleSAML\SAML2\XML\%s\%s', $prefix, $eltName);
```

Comment:
I was thinking about creating a utility method to look up known classes, so we can re-use it elsewhere. we shouldn't be using Chunks if we know the element, but we still do that in many places in the lib. The utility method could also query the class-registry to be able to recognize classes outside this lib.

Reply:
Didn't we create a registry in the container for custom handlers for given XML elements? In any case, I agree we should do that everywhere instead of just using `Chunk`.